scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -959,7 +960,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           Type.Macro(macroSplice())
         case _: Underscore if inMatchType =>
           next(); Type.PatWildcard()
-        case _: Underscore if dialect.allowUnderscoreAsTypePlaceholder =>
+        case _: Underscore
+            if dialect.allowUnderscoreAsTypePlaceholder ||
```

Comment:
@tgodzik this parameter is currently added to Scala3Future dialect and not Scala33 or earlier. I wonder if it has in fact already been released, which is why the submitter sees this problem in his Scala33 codebase. I tried searching dotty release notes but couldn't find anything since 2021.

Reply:
This never happened :/ It was supposed to happen in 3.2, but now the site is heavily outdated 

https://docs.scala-lang.org/scala3/reference/changed-features/wildcards.html